src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -774,8 +774,14 @@ private Task OnRuntimeInitializeStop(CancellationToken ct)
                     // Write "ShutDown" state in the table + broadcast gossip msgs to re-read the table to everyone
                     await scheduler.QueueTask(this.membershipOracle.ShutDown, this.membershipOracleContext)
                         .WithTimeout(stopTimeout, $"MembershipOracle Shutting down failed due to timeout {stopTimeout}");
+                    // Awful, but some components queue a task when receiving silo status notification,
+                    // add a delay here to give them a chance to execute this task
```

Comment:
does this call for including those components shutdown logic into lifecycle so we don't need to do this hack?